external/java/src/main/java/ch/epfl/dedis/calypso/CalypsoRPC.java

```diff
@@ -42,8 +42,7 @@ public CalypsoRPC(ByzCoinRPC byzcoin, DarcId darcId, Roster ltsRoster, List<Sign
         LTSInstance inst = new LTSInstance(this, darcId, ltsRoster, signers, signerCtrs);
         Proof proof = inst.getProof();
         // Start the LTS/DKG protocol.
-        CreateLTSReply lts = createLTS(proof);
-        this.lts = lts;
+        this.lts = createLTS(proof);
```

Comment:
I'm not sure, but I don't see where proof.verify() is ever called here.

Reply:
You're right, it's not just at this place, the proof is not checked in some other places too. I'm looking for places that calls `getProof` and adding the verification for those that need it.

Reply:
hopefully this commit https://github.com/dedis/cothority/commit/fed20e7684a893f30702ee2de2d0ecbebb6aa94f addresses this issue in general. in the new code, the proof is always verified during construction because there is no reason (under normal operation) that the forwardlinks are wrong or the Merkle-root is wrong. the `exists` and `match` functions are staying the same. they are only called when it's necessary.